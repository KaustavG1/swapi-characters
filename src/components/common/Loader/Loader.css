.loader-page {
  height: 85vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.loader {
  font-size: 5rem;
}
