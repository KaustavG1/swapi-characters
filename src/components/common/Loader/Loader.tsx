import "./Loader.css";

function Loader() {
  return (
    <div className="loader-page">
      <div className="loader">Loading ...</div>
    </div>
  );
}

export default Loader;
