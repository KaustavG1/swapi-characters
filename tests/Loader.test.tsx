import { it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Loader from "../src/components/common/Loader/Loader";

describe("Loader component", () => {
  it("renders 'Loader' component", () => {
    const { container } = render(<Loader />);
    const loading = screen.getByText(/Loading .../i);
    expect(loading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
