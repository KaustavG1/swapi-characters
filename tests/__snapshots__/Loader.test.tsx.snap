// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyRecords component > renders 'EmptyRecords' component 1`] = `
<div>
  <div
    class="blank-page"
  >
    <div
      class="loader"
    >
      Loading ...
    </div>
  </div>
</div>
`;

exports[`Loader component > renders 'Loader' component 1`] = `
<div>
  <div
    class="blank-page"
  >
    <div
      class="loader"
    >
      Loading ...
    </div>
  </div>
</div>
`;
